lib/domain/services/collection_services.dart

```diff
@@ -1,15 +1,69 @@
 import 'package:memo/data/repositories/collection_repository.dart';
+import 'package:memo/data/repositories/memo_repository.dart';
+import 'package:memo/domain/isolated_services/memory_stability_services.dart';
 import 'package:memo/domain/models/collection.dart';
+import 'package:memo/domain/transients/collection_status.dart';
 
+/// Handles all domain-specific operations pertaining to one or multiple [Collection]
 abstract class CollectionServices {
+  /// Retrieves all available [CollectionStatus] and keeps listening to any changes made to them
+  Future<List<Stream<CollectionStatus>>> listenToAllCollectionsStatus();
+
+  /// Retrieves all available [Collection]
   Future<List<Collection>> getAllCollections();
+
+  /// Retrieves the [Collection] (with [collectionId]) current memory stability
```

Comment:
I would like to suggest a doc pattern that we could follow: we could refer to the id properties as it were their own entities

Example:
`Retrieves the [collectionId] current memory stability`

I noticed that in most of the situations, the id property has the name of its entity, so we could refer to it as it was the entity itself

Reply:
The problem here is that it's hard to determine when a function signature may be implied that an `{String id}` belongs to that particular service-oriented entity. For instance, there are a fetch in memoservices that returns all memos, but one with the collection Id and one that return a single one with the unique id for that particular memo. In this case, it clearly doesn't specify that the ID belongs collection, so the argument being more verbose is intended, otherwise, I would say that this signature should be `getCollectionMemoryStabilityByCollectionId(String id)` which, in my opinion is a little bit worse, but debatable